core/plugins/org.polarsys.capella.core.platform.sirius.sirius.validation/src/org/polarsys/capella/core/platform/sirius/sirius/validation/ddiagram/CapellaElementInDescriptionExistanceCheck.java

```diff
@@ -63,17 +94,115 @@ public void startElement(String uri, String localName, String qName, Attributes
               && attName.equalsIgnoreCase(IConstantValidation.XHTML_HREF_ATT)) {
             // get id
             // get element
-            EObject eObject = SaxParserHelper.getEObjectFromHrefAttribute(capellaElement, attValue);
+            EObject eObject = SaxParserHelper.getEObjectFromHrefAttribute(element, attValue);
 
             if (null == eObject) {
               // element does not exist in the resource
-              result[0] = ctx.createFailureStatus();
+              elementId = attValue.replace("hlink://", "");
+              isElementExist = false;
               break;
             }
           }
         }
       }
     }
+
+    @Override
+    public void characters(char ch[], int start, int length) throws SAXException {
+      String string = new String(ch, start, length);
+      elementValue.append(string);
+    }
+
+    @Override
+    public void endElement(String uri, String localName, String qName) throws SAXException {
+      // a tag
+      if ((qName.equalsIgnoreCase(IConstantValidation.XHTML_A_TAG)) && !isElementExist) {
+        String value = elementValue.toString();
+        value = value.replaceAll("\\s+", " "); //$NON-NLS-1$//$NON-NLS-2$
+        String elementName = getName(element);
+        if (!alreadyParsedLinks.contains(elementId)) {
+          alreadyParsedLinks.add(elementId);
+          String failureMessage = "(Hyperlink) The model/diagram element named �" + value + "� (id: " + elementId
+              + ") can not be found for the rich text description of the element " + elementName;
+          result.add(ConstraintStatus.createStatus(ctx, element, ctx.getResultLocus(), "{0}", failureMessage));
+        } else {
+          List<IStatus> updatedResult = result.stream().map(sts -> {
+            if (sts.getMessage().contains(elementId)) {
+              String name = extractName(sts.getMessage());
+              String failureMessage = "(Hyperlink) The model/diagram elements named �" + name + ", ...� (id: "
+                  + elementId + ") can not be found for the rich text description of the element " + elementName;
+              return ConstraintStatus.createStatus(ctx, element, ctx.getResultLocus(), "{0}", failureMessage);
+            }
+            return sts;
+          }).collect(Collectors.toList());
+          result.clear();
+          result.addAll(updatedResult);
+        }
+        // re-init for new element to be found or not
+        isElementExist = true;
+        // empty the value and the id
+        elementValue = new StringBuilder();
+        elementId = "";
+      }
+    }
+
+  }
+
+  public static String getName(EObject object) {
```

Comment:
First, code duplication with CapellaElementInDescriptionNameCheck . Second, look for an existing method that gives the element name, if not exist, contribute in one of the generic helper and use here.